benchmark.py

```
@@ -0,0 +1,833 @@
+
+async def insert_batch(session: aiohttp.ClientSession, url: str, batch: List[Dict[str, Any]]) -> None:
+    """Helper function to insert a batch of fingerprints."""
+    if not batch:
+        return
+    content = {'changes': batch}
+    async with session.post(url, json=content) as response:
+        response.raise_for_status()
+
+
+async def run_insertion_mode(session: aiohttp.ClientSession, url: str, 
+                            fingerprint_generator, num_docs: int, 
```

Ruff (0.12.2)

Missing type annotation for function argument `fingerprint_generator`

(ANN001)